nym-vpn/ui/src/pages/settings/Settings.tsx

```diff
@@ -2,61 +2,221 @@ import { useEffect, useState } from 'react';
-    state.uiTheme === 'Dark',
-  );
+  const [entrySelector, setEntrySelector] = useState(state.entrySelector);
+  const [autoConnect, setAutoConnect] = useState(state.autoConnect);
+  const [monitoring, setMonitoring] = useState(state.monitoring);
 
   useEffect(() => {
-    setDarkModeEnabled(state.uiTheme === 'Dark');
+    setEntrySelector(state.entrySelector);
+    setAutoConnect(state.autoConnect);
+    setMonitoring(state.monitoring);
   }, [state]);
 
-  const handleThemeChange = async (darkMode: boolean) => {
-    if (darkMode && state.uiTheme === 'Light') {
-      dispatch({ type: 'set-ui-theme', theme: 'Dark' });
-    } else if (!darkMode && state.uiTheme === 'Dark') {
-      dispatch({ type: 'set-ui-theme', theme: 'Light' });
-    }
-    invoke<void>('set_ui_theme', { theme: darkMode ? 'Dark' : 'Light' }).catch(
-      (e) => {
-        console.log(e);
-      },
-    );
+  const handleEntrySelectorChange = async () => {
+    const isSelected = !state.entrySelector;
+    dispatch({ type: 'set-entry-selector', entrySelector: isSelected });
+    invoke<void>('set_entry_location_selector', {
+      entrySelector: isSelected,
+    }).catch((e) => {
+      console.log(e);
+    });
+  };
+
+  const handleAutoConnectChanged = async () => {
+    const isSelected = !state.autoConnect;
+    dispatch({ type: 'set-auto-connect', autoConnect: isSelected });
+    invoke<void>('set_auto_connect', { autoConnect: isSelected }).catch((e) => {
+      console.log(e);
+    });
+  };
+
+  const handleMonitoringChanged = async () => {
+    const isSelected = !state.monitoring;
+    dispatch({ type: 'set-monitoring', monitoring: isSelected });
+    invoke<void>('set_monitoring', { monitoring: isSelected }).catch((e) => {
+      console.log(e);
+    });
   };
 
   return (
-    <div className="h-full flex flex-col p-4 gap-4">
-      <div className="flex flex-row justify-between items-center">
-        <p className="text-lg text-baltic-sea dark:text-mercury-pinkish">
-          Dark Mode
-        </p>
-        <Switch
-          checked={darkModeEnabled}
-          onChange={handleThemeChange}
-          className={clsx([
-            darkModeEnabled
-              ? 'bg-melon'
-              : 'bg-mercury-pinkish dark:bg-gun-powder',
-            'relative inline-flex h-6 w-11 items-center rounded-full',
-          ])}
-        >
-          <span className="sr-only">Dark mode</span>
-          <span
-            className={clsx([
-              darkModeEnabled ? 'translate-x-6' : 'translate-x-1',
-              'inline-block h-4 w-4 transform rounded-full bg-cement-feet dark:bg-white transition',
-            ])}
-          />
-        </Switch>
+    <div className="h-full flex flex-col p-4 mt-2 gap-8">
+      <SettingsGroup
+        settings={[
+          {
+            title: t('auto-connect.title'),
+            desc: t('auto-connect.desc'),
+            leadingIcon: 'hdr_auto',
+            onClick: () => {},
```

Comment:
you can make `onClick` props optional in `SettingsGroup`, so you dont need to pass a dummy function